### PR TITLE
chore(trie): fuzz test for Put and Get

### PR DESCRIPTION
**Pull request by qdm12**
## Changes

- Change one unit test to a fuzz test
- Add CI workflow

## Tests

```
go test -fuzz=Fuzz_Trie_PutAndGet -fuzztime=10s github.com/ChainSafe/gossamer/lib/trie
```

## Issues

- #2087 

## Primary Reviewer

- Anyone :tada:

### CODE REVIEW

**Comment by EclesioMeloJunior** [comment]
Should we change the name of the CI file to `fuzz_trie` as it only executes the `Fuzz_Trie_PutAndGet` test?

**Comment by EclesioMeloJunior** [comment]
@qdm12 ok that makes sense if we plan to add more fuzz tests there

**Review by EclesioMeloJunior — APPROVED** [review]

File: .github/workflows/fuzz.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+      - name: Go modules cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.go-cache-paths.outputs.go-mod }}
+          key: ${{ runner.os }}-go-mod-${{ hashFiles('**/go.sum') }}
+          restore-keys: ${{ runner.os }}-go-mod
+
+      - name: Fuzz trie
+        run: go test -fuzz=Fuzz_Trie_PutAndGet -fuzztime=5m github.com/ChainSafe/gossamer/lib/trie
```
Comment (timwu20) [inline .github/workflows/fuzz.yml:39]: couldn't we do `go test -fuzz=Fuzz* -fuzztime=5m ./...` to just automatically fuzz everything

Comment (qdm12) [reply .github/workflows/fuzz.yml:39]: No you actually can't do that (I was surprised too).

i.e. `cannot use -fuzz flag with multiple packages`

**Review by timwu20 — APPROVED** [review]

**Pull request merged**